consensus/src/epoch_manager.rs

```diff
@@ -419,7 +435,17 @@ impl EpochManager {
 
         let safety_rules_container = Arc::new(Mutex::new(safety_rules));
 
-        self.commit_state_computer.new_epoch(&epoch_state);
+        let (consensus_to_quorum_store_sender, consensus_to_quorum_store_receiver) =
+            mpsc::channel(self.config.intra_consensus_channel_buffer_size);
+        self.spawn_quorum_store(consensus_to_quorum_store_receiver);
+        let payload_manager = QuorumStoreClient::new(
+            consensus_to_quorum_store_sender.clone(),
+            self.config.quorum_store_poll_count,
+            self.config.quorum_store_pull_timeout_ms,
+        );
+
+        self.commit_state_computer
+            .new_epoch(&epoch_state, consensus_to_quorum_store_sender);
```

Comment:
State computer uses the sender to update Quorum Store with the latest epoch and round. This is required for proper garbage collection for a proper Quorum Store (it's not used in this Direct Mempool Quorum Store).

Since state computer outlives epochs, we need to supply a new sender for each epoch. It makes new_epoch even more gross :-/

Reply:
yeah it feels a little overwhelmed